### DISCUSSION TITLE
Running on AMD gpu on Windows without installing the full SDK

### DISCUSSION THREAD

**Discussion by DGdev91** [leejet/stable-diffusion.cpp#194] [Show and tell]
The ROCm version of KoboldCpp (https://github.com/YellowRoseCx/koboldcpp-rocm) copies it's runtime libs in a temporary folder called "_MEI" + some nubers, placed in C:\Users\username\AppData\Local\Temp. In those libs there are also hipblas.dll, rocblas.dll, and a "library" folder containing ROCm's tensile library files, and this allows it to run on amd gpus without the need of having the SDK installed.

I just checked, and this works just fine for stablediffussion.cpp too. just copy those libs from koboldcpp's temp folder and place it in the same place as sd.exe

Maybe i can also be a good idea shipping those libs in the rocm windows build

**Comment by stduhpf** [comment]
Bless you! I was getting realy skill issued trying to get my ROCm build to link properly to the SDK, and this solved it for me, though things have changed a bit since you started that discussion.

I couldn't find `rocblas.dll` in the "_MEI" temp folder, only `hipblas.dll`, however in the more recent relseases of the ROCm version of KoboldCpp, there are zip files alongside the exe ([koboldcpp_rocm_files.zip](https://github.com/YellowRoseCx/koboldcpp-rocm/releases/download/v1.95.1.yr0-ROCm/koboldcpp_rocm_files.zip)), which contains some of the libraries used by koboldcpp, including `rocblas.dll`.
